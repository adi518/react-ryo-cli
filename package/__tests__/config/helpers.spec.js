const { SCRIPTS } = require("../../config/scripts.js");
const {
  Args,
  preflight,
  getLibraryName,
  normalizeScript
} = require("../../config/helpers.js");

console.log = () => {}; // eslint-disable-line
console.error = () => {}; // eslint-disable-line

describe("preflight", () => {
  it("should return falsy for eject script", () => {
    expect(preflight({ script: "eject" })).toBe(false);
  });

  it("should return truthy for verified scripts", () => {
    const VERIFIED_SCRIPTS = { ...SCRIPTS };
    delete VERIFIED_SCRIPTS.EJECT;

    const results = Object.values(VERIFIED_SCRIPTS)
      .map(script => preflight({ script }))
      .every(result => result);

    expect(results).toBe(true);
  });
});

describe("getLibraryName", () => {
  it("should return in pascal case", () => {
    expect(getLibraryName({ name: "foo-bar" })).toMatchSnapshot();
  });
});

describe("Args class", () => {
  it("should prepend Craco binary argument", () => {
    expect(new Args("foo").prependCracoBin()).toMatchSnapshot();
  });

  it("should append Craco configuration arguments", () => {
    expect(new Args("foo").appendCracoConfig()).toMatchSnapshot();
  });

  it("should add Craco arguments", () => {
    expect(new Args("foo").addCraco()).toMatchSnapshot();
  });
});

describe("normalizeScript", () => {
  test("build", () => {
    expect(normalizeScript({ rawScript: SCRIPTS.BUILD })).toMatchSnapshot();
  });

  test("build:production", () => {
    expect(
      normalizeScript({ rawScript: SCRIPTS.BUILD_PRODUCTION })
    ).toMatchSnapshot();
  });

  test("build:package", () => {
    expect(
      normalizeScript({ rawScript: SCRIPTS.BUILD_PACKAGE })
    ).toMatchSnapshot();
  });

  test("build:package:production", () => {
    expect(
      normalizeScript({ rawScript: SCRIPTS.BUILD_PACKAGE_PRODUCTION })
    ).toMatchSnapshot();
  });

  test("build:stats", () => {
    expect(
      normalizeScript({ rawScript: SCRIPTS.BUILD_STATS })
    ).toMatchSnapshot();
  });

  test("test", () => {
    expect(normalizeScript({ rawScript: SCRIPTS.TEST })).toMatchSnapshot();
  });

  test("test:watch", () => {
    expect(
      normalizeScript({ rawScript: SCRIPTS.TEST_WATCH })
    ).toMatchSnapshot();
  });

  test("test:update", () => {
    expect(
      normalizeScript({ rawScript: SCRIPTS.TEST_UPDATE })
    ).toMatchSnapshot();
  });

  test("test:production", () => {
    expect(
      normalizeScript({ rawScript: SCRIPTS.TEST_PRODUCTION })
    ).toMatchSnapshot();
  });

  test("eject", () => {
    expect(normalizeScript({ rawScript: SCRIPTS.EJECT })).toMatchSnapshot();
  });
});
