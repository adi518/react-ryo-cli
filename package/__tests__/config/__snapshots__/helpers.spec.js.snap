// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CracoArgs class should add Craco arguments 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "foo",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`CracoArgs class should append Craco configuration arguments 1`] = `
Array [
  "foo",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`CracoArgs class should prepend Craco binary argument 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "foo",
]
`;

exports[`getLibraryName should return in pascal case 1`] = `"FooBar"`;

exports[`normalizeScript build 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "build",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`normalizeScript build:package 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "build",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`normalizeScript build:package:production 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "build",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`normalizeScript build:production 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "build",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`normalizeScript build:stats 1`] = `
Array [
  "./node_modules/source-map-explorer/dist/cli.js",
  "out/static/js/*.js",
]
`;

exports[`normalizeScript eject 1`] = `"eject"`;

exports[`normalizeScript test 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "test",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`normalizeScript test:production 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "test",
  "--ci",
  "--collectCoverage",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`normalizeScript test:update 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "test",
  "--updateSnapshot",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;

exports[`normalizeScript test:watch 1`] = `
Array [
  "./node_modules/@craco/craco/bin/craco.js",
  "test",
  "--config",
  "config\\\\craco\\\\craco.config.js",
]
`;
